diff --git a/CMakeLists.txt b/CMakeLists.txt
index 030cacc..34e9bf9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ target_compile_features(${PROJECT_NAME}
     INTERFACE cxx_std_17
   )
 
-target_link_libraries(${PROJECT_NAME} INTERFACE OpenMP::OpenMP_CXX)
+target_link_libraries(${PROJECT_NAME} INTERFACE OpenMP::OpenMP_CXX rt)
 
 option(QoZ_USE_BUNDLED_ZSTD "prefer the bundled version of Zstd" OFF)
 option(QoZ_DEBUG_TIMINGS "print debug timing information" ON)
@@ -44,11 +44,11 @@ endif()
 pkg_search_module(ZSTD IMPORTED_TARGET libzstd)
 if(ZSTD_FOUND AND NOT QoZ_USE_BUNDLED_ZSTD)
   message("Zstd found.")
-  target_link_libraries(${PROJECT_NAME} INTERFACE PkgConfig::ZSTD)
+  target_link_libraries(${PROJECT_NAME} INTERFACE PkgConfig::ZSTD rt)
 else()
   message("Zstd not found.")
   add_subdirectory(zstd)
-  target_link_libraries(${PROJECT_NAME} INTERFACE zstd)
+  target_link_libraries(${PROJECT_NAME} INTERFACE zstd rt)
 endif()
 
 find_package(GSL)
